training/training_code/train_ner.py

```diff
@@ -628,6 +628,15 @@ def ftags(name):
     # train confidence branch
     ner_model = train_confidence_branch(ner_model, params, train_data, valid_data, ckpt_dir)
 
+    # generate entity-level metrics for training data
```

Comment:
Shall we just compute for `train`, `validation` and `test` then?

i.e. `for dataset in ['train', 'valid', 'test']:` ?

Reply:
I also need to update the metrics computation to be entity-level rather than tag-level

Reply:
I think test data is usually inaccessible to the user during the development, and should only be used in evaluation? (I believe the `test_` variables in this script actually refer to validation?)

Reply:
One would expect anyone training to have a train, valid and test set. Here it refer to test set.

Hence why if we will compute metrics for training, we may as well compute for all 3 sets

Reply:
Got it, thought on something different. Let me change the script